lib/rules/no-misleading-character-class.js

```diff
@@ -159,20 +175,23 @@ module.exports = {
             }
 
             const foundKinds = new Set();
```

Comment:
```suggestion
            const foundKinds = [];
```

A set is no longer needed now that all elements are distinct.